src/App.tsx

```diff
@@ -2,108 +1,69 @@
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
 
 export const App: React.FC = () => {
+  const totalItems: number = 42;
+  const values = [3, 5, 10, 20];
+  const [numItems, setNumItems] = useState(5);
+
+  const [currentPage, setCurrentPage] = useState(1);
+
+  let from = 1;
+  let to = numItems;
+
+  if (currentPage) {
+    from = (currentPage - 1) * numItems + 1;
+    to = Math.min(currentPage * numItems, totalItems);
+  }
+
+  const items = getNumbers(from, to).map(n => `Item ${n}`);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${from} - ${to} of ${totalItems})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            className="form-control"
+            value={numItems}
+            onChange={event => {
```

Comment:
Converting the event target value to a number is good for ensuring the correct data type, but consider validating the value to ensure it is within the expected range of `values`.